File: library/src/main/java/com/getstream/sdk/chat/rest/apimodel/response/ChannelUserRead.java
package com.getstream.sdk.chat.rest.apimodel.response;

import com.getstream.sdk.chat.model.User;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class ChannelUserRead {
    @SerializedName("user")
    @Expose
    private User user;

    @SerializedName("last_read")
    @Expose
    private String last_read;

    public User getUser() {
        return user;
    }

    public String getLast_read() {
        return last_read;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public void setLast_read(String last_read) {
        this.last_read = last_read;
    }
}

File: app/src/main/java/io/getstream/chat/android/core/poc/library/ChannelUserRead.kt
package io.getstream.chat.android.core.poc.library

import com.google.gson.annotations.Expose

import com.google.gson.annotations.SerializedName
import java.util.*


class ChannelUserRead : UserEntity {

    @SerializedName("user")
    
    lateinit var user: User

    @SerializedName("last_read")
    
    var lastRead = Date()

    constructor(user: User, lastRead: Date) {
        this.user = user
        this.lastRead = lastRead
    }


    override fun getUserId(): String {
        return user.id
    }
}